taar/recommenders/collaborative_recommender.py

```diff
@@ -89,22 +90,25 @@ def recommend(self, client_data, limit, extra_data={}):
         # space.
         distances = {}
         for addon in self.raw_item_matrix:
-            # We don't really need to show the items we requested. They will always
-            # end up with the greatest score. Also filter out legacy addons from the
-            # suggestions.
-            hashed_id = str(addon.get("id"))
-            if (hashed_id in installed_addons or
-                    hashed_id not in self.addon_mapping or
-                    self.addon_mapping[hashed_id].get("isWebextension", False) is False):
+            # We don't really need to show the items we requested.
+            # They will always end up with the greatest score. Also
+            # filter out legacy addons from the suggestions.
+            hashed_id = positive_hash(addon.get("id"))
+            str_hashed_id = str(hashed_id)
+            if (hashed_id in installed_addons_as_hashes or
```

Comment:
Oh, sorry, this slept in before: looks like you're now checking for `hashed_id` and not for `str(hashed_id)` to be in `installed_addons_as_hashes`. Was this a bug before or is this changing the behaviour now? If it was a bug, would you mind adding a test case? If it's not, why is this being changed?

Reply:
I'm pretty sure this was a bug.  I modified the fixture data in the testcase to key the FAKE_MAPPING dictionary using the `positive_hash` function to match up to the comment at the top of the `CollaborativeRecommender::recommend` method that indicates that the model expects positive java hashes.

The old testcase only checked that a recommendation was made - not that the recommendation was necessarily a good one.  I added `test_best_recommendation` to assert the expected recommendations are correct which also tests that the dictionary lookups are correct.